### PR TITLE
Delete attachments of Cards in Folders and Lists

### PR DESCRIPTION
**Pull request by drparcon**
This pull requests solves the issue of attachments of cards in folders and lists not being deleted when the respective folder or list is deleted. This is handled by having a respective endpoint for folder and list and fetching that endpoint when a list or a folder is deleted. The database function for deleting a folder parses the folder's cards and calls the function to delete the attachments of that card. The database function for deleting a list parses the elements of a list and calls the function to delete attachments of a card if it finds a card and calls the function for delete folder attachments if it finds a folder.

### CODE REVIEW

File: src/components/List.svelte [hunk 1]
```diff
@@ -39,6 +39,15 @@
...
+    fetch(`/api/card/attachment/delete/list/${list._id}`, {
+      method: "DELETE",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        user_id: $session.user_id,
+      }),
+    });
     $session.lists = $session.lists.filter((l) => {
```
Comment (grantyap) [inline src/components/List.svelte:51]: Please add a new line here to visually separate the `fetch` from the rest of the code.

```suggestion
    });
    
    $session.lists = $session.lists.filter((l) => {
```

File: src/routes/api/_database.js [hunk 2]
```diff
@@ -130,6 +130,44 @@ export async function deleteAttachmentsOfCard(cardId) {
   return result;
 }
 
+//TODO: Find a way to make Mongo do the parsing and deleting work
+export async function deleteAttachmentsinFolder(userId, folderId) {
```
Comment (grantyap) [inline src/routes/api/_database.js:134]: Please capitalize each succeeding word in function names.

```suggestion
export async function deleteAttachmentsInFolder(userId, folderId) {
```

File: src/routes/api/_database.js [hunk 3]
```diff
@@ -130,6 +130,44 @@ export async function deleteAttachmentsOfCard(cardId) {
...
+  }
+
+  for (const card of folder.cards) {
+    deleteAttachmentsOfCard(card._id);
+  }
+
+  return;
+}
+
+export async function deleteAttachmentsinList(userId, listId) {
```
Comment (grantyap) [inline src/routes/api/_database.js:153]:
```suggestion
export async function deleteAttachmentsInList(userId, listId) {
```

**Review by grantyap — CHANGES_REQUESTED** [review]

**Review by BigBlackLab19 — APPROVED** [review]

File: src/routes/api/_database.js [hunk 4]
```diff
@@ -148,6 +148,42 @@ export async function deleteAttachmentsOfCard(cardId) {
...
+        if (element.card_name) {
+          deleteAttachmentsOfCard(element._id);
+        } else if (element.folder_name) {
+          deleteAttachmentsInFolder(userId, element._id);
+        }
+      }
+    }
+  }
+
+  return;
```
Comment (grantyap) [inline src/routes/api/_database.js:184]: Please remove this `return` statement also.

**Review by grantyap — CHANGES_REQUESTED** [review]
Looks good so far!

Last thing is to get rid of the lint errors in AddCalendar.

**Comment by drparcon** [comment]
> 
> 
> One more minor change and this will be good to go. Last thing is to get rid of the lint errors in AddCalendar.

The same lint errors also exist in the development branch. I think it would be better to take care of them there before merging this branch into development.

**Pull request merged**